### DISCUSSION TITLE
@nivo/line gridYValues prop usage

### DISCUSSION THREAD

**Discussion by krau5** [plouc/nivo#2012] [Q&A] [answered]
Hey there :)

**Description**
I'm trying to use `@nivo/line`. The svg variant, to be precise.
I want to make custom `gridYValues` but it doesn't actually work.

**Code**
```
<Line
  data={data}
  gridYValues={[2000, 4000, 6000, 8000]}
/>
```

**Question & additional description**
`data` y values are in range from 3750 to 6500.
Is it only possible using `ResponsibleLineCanvas` instead of `ResponsiveLine`?

Please, let me know if I forgot to specify something.
Thanks for the attention :)

**Comment by tkonopka** [comment]
Hi @krau5. Setting custom gridYValues works for me with the svg Line component. Your code format looks alright, except that if you're using Line and not ResponsiveLine, also specify the width and height for the chart. 

If you're still having trouble, can you post a more complete example?

**Comment by krau5** [comment]
Thanks for the tip @tkonopka. The `<Line />` component has a lil' bit more complex structure.

```
// Line component

import { useTheme } from '@nivo/core';
import { ResponsiveLine, LineSvgProps } from '@nivo/line';

type Props = LineSvgProps & {
  width: number;
  height: number;
};

export const Line = ({ theme: customTheme, ...props }: Props) => {
  const theme = useTheme();

  return <ResponsiveLine {...props} theme={{ ...theme, ...customTheme }} />;
};
```

And the usage looks like:

```
<Line
  data={data}
  width={900}
  height={400}
  gridYValues={[2000, 4000, 6000, 8000]}
/>
```

P.S. I've just changed it from usual `Line` to `ResponsiveLine`, although it doesn't fix the problem

**Reply by tkonopka** [reply]
Thanks for pasting more of the context.

The name 'Line' is used by nivo to denote a component that draws an svg chart of a fixed size, but here you are defining a different object with that same name. To avoid confusion, let's call your component 'CustomLine'. Your component is really a ResponsiveLine (with some extras). It will therefore take its size from its parent container. Try something like this:

```
<div style={{ width: 900, height: 400 }}>
  <CustomLine
    data={data}
    margin={{ top: 60, right: 60, bottom: 60, left: 60 }}
    gridYValues={[2000, 4000, 6000, 8000]}
  />
</div>
```

This should display a chart with axes, and from there you can tune the position of grid lines, adjust the size of the parent div, etc.

https://codesandbox.io/s/eager-morning-5mgxjc?file=/src/App.js

**Reply by krau5** [reply]
The sandbox you have provided doesn't support `gridYValues` as well 😅 
Any of the values I have passed ( `[17, 37, 117, 177]` as an example to check ) are not shown.
Should it be declared as a bug?

UPD. I've also adjusted it to the example you have written above, although, it hasn't helped a lot :(

**Reply by tkonopka** [reply]
Sorry to hear. That's puzzling as it seems to work for me.

<img src="https://user-images.githubusercontent.com/7260190/169784744-46402ea8-2da3-487e-a21c-c52545de8bff.png" width=400 />

Perhaps someone else can pitch in and point to what might be missing.

**Reply by plouc** [reply]
@krau5, I don't know what you're seeing, but it works:
<img width="416" alt="Screen Shot 2022-05-23 at 18 05 46" src="https://user-images.githubusercontent.com/501642/169784725-9d4e53ae-71a1-443b-8885-e741b220e083.png">
There are grid lines on the Y axis for `[20, 40, 120, 180]` (and works with `[17, 37, 117, 177]` as well ).

Maybe you're referring to the ticks on the left axis? In which case, this would be:

```
axisLeft={{ tickValues: [17, 37, 117, 177] }}
```

Just a guess.

**Reply by krau5** [reply]
Okay :( All in all, thanks for a help <br />
<img width="1352" alt="image" src="https://user-images.githubusercontent.com/52201020/169785412-f010b47b-c7b3-46e0-b0a3-35a599eb978f.png">

**Reply by krau5** [reply]
> @krau5, I don't know what you're seeing, but it works: <img alt="Screen Shot 2022-05-23 at 18 05 46" width="416" src="https://user-images.githubusercontent.com/501642/169784725-9d4e53ae-71a1-443b-8885-e741b220e083.png"> There are grid lines on the Y axis for `[20, 40, 120, 180]` (and works with `[17, 37, 117, 177]` as well ).
> 
> Maybe you're referring to the ticks on the left axis? In which case, this would be:
> 
> ```
> axisLeft={{ tickValues: [17, 37, 117, 177] }}
> ```
> 
> Just a guess.

Will have a look

**Reply by krau5** [reply]
@plouc thanks for a tip :) Mistaken in explaining the main goal

**Comment by krau5** [comment, marked as answer]
Solution:
https://github.com/plouc/nivo/discussions/2012#discussioncomment-2802976
